doc/tutorials/others/barcode_detect_and_decode.markdown

```diff
@@ -0,0 +1,76 @@
+Barcode Recognition    {#tutorial_barcode_detect_and_decode}
+===================
+
+@tableofcontents
+
+@prev_tutorial{tutorial_traincascade}
+@next_tutorial{tutorial_introduction_to_svm}
+
+|    |    |
+| -: | :- |
+| Compatibility | OpenCV >= 4.8 |
+
+Goal
+----
+
+In this chapter we will familiarize with the barcode detection and decoding methods available in OpenCV.
+
+Basics
+----
+
+Barcode is major technique to identify commodity in real life. A common barcode is a pattern of parallel lines arranged by black bars and white bars with vastly different reflectivity. Barcode recognition is to scan the barcode in the horizontal direction to get a string of binary codes composed of bars of different widths and colors, that is, the code information of the barcode. The content of barcode can be decoded by matching with various barcode encoding methods. Currently, we support EAN-8, EAN-13, UPC-A and UPC-E standards.
+
+See https://en.wikipedia.org/wiki/Universal_Product_Code and https://en.wikipedia.org/wiki/International_Article_Number
+
+Related papers: @cite Xiangmin2015research , @cite kass1987analyzing , @cite bazen2002systematic
+
+Code example
+------------
+
+### Main class
+Several algorithms were introduced for barcode recognition.
+
+While coding, we firstly need to create a cv::barcode::BarcodeDetector object. It has mainly three member functions, which will be introduced in the following.
+
+#### Initialization
+
+User can construct BarcodeDetector with super resolution model which should be downloaded automatically to `<opencv_build_dir>/downloads/barcode`. If not, please download them from https://github.com/WeChatCV/opencv_3rdparty/tree/wechat_qrcode or choose not to use super resolution.
+
+@snippet ./samples/barcode.cpp initialize
```

Comment:
Barcode sample is located in wrong place: `modules/objdetect/samples/barcode.cpp`, but should be in `samples/cpp/barcode.cpp`.

Reply:
Technically OpenCV supports module-specific samples (like in OpenCV contrib). I decided to put it there because it demonstrates `objdetect` module functionality. Perhaps we can continue moving C++ samples to their respective modules to reduce this list: https://github.com/opencv/opencv/blob/19f4f2eb92313cfe688ba924a784fa775c99ae4c/samples/cpp/CMakeLists.txt#L3-L21
If you think it is better to move it to others (`samples/cpp`) I will do it.

Reply:
It's very unusual for main opencv repo and users do not search for samples in modules source code. I propose to put it to expected places and introduce new practice later globally.

Reply:
Ok, done!